src/Npgsql/Schema/DbColumnSchemaGenerator.cs

```diff
@@ -150,7 +150,8 @@ internal ReadOnlyCollection<NpgsqlDbColumn> GetColumnSchema()
                     populatedColumns++;
                 }
 
-                result[i]!.ColumnName = result[i]!.BaseColumnName = field.Name.StartsWith("?column?") ? null : field.Name;
+                result[i]!.BaseColumnName ??= field.Name.StartsWith("?column?") ? null : field.Name;
+                result[i]!.ColumnName = field.Name.StartsWith("?column?") ? null : field.Name;
```

Comment:
Looks good, but let's check the column name once and an assertion to eliminate bangs:
```suggestion
                Debug.Assert(result != null);

                var fieldName = field.Name.StartsWith("?column?") ? null : field.Name;
                
                result[i].BaseColumnName ??= fieldName;
                result[i].ColumnName = fieldName;
```

Reply:
Thanks for reviewing @YohDeadfall 

Changes committed